app/javascript/utils/fieldSpecs.js

```diff
@@ -15,10 +15,16 @@ const getRenderType = (value) => {
   }
 }
 
+const dateIsJson = (date) => {
+  return (typeof date === 'object' && 'month' in date && 'day' in date && 'year' in date)
+}
+
 export const formatValue = (value, type) => {
```

Comment:
this is the issue that came up in the test when I had updated the domain application to have it be an object.